app/assets/javascripts/utilities/buildCommentHTML.js.erb

```diff
@@ -95,10 +95,9 @@ function actions(comment) {
 
 function reactions(comment) {
   if (comment.newly_created) {
-    return '<button class="reaction-button reacted" id="button-for-comment-'+comment.id+'" data-comment-id="'+comment.id+'">\
```

Comment:
**What is  this change doing?**

The BE does **not** treat newly created comments as "reacted" so it is misleading when we render it as such. It can also force the BE to get into weird states (which is another issue altogether...)

Reply:
> There's an after_create_commit callback here that executes the following:

Ah! I see.

Is there a way to make this transactional such that the comment and reaction are created at the same time? Is that a change you think will be helpful?

There's a small performance tradeoff, but I imagine creating a reaction is a pretty small task that it's negligible :shrug:

Reply:
At the moment I think adding better handling of the AJAX response when reacting could fix the issue. The JSON payload includes `"result"` which is either `create`/`destroy`. This means a new reaction is created or an existing reaction is destroyed, there's no in between [in the controller](https://github.com/fdoxyz/forem/blob/master/app/controllers/reactions_controller.rb#L68-L80).

What this means is that even if the race condition happens in a production environment (i.e. high load on work queue) and two reactions are created, once they're both created then the destroy should happen twice and it should go back to normal (no more than 1 reaction from the comment author).

I will get back to you on this when I get a chance to dive a bit deeper on the topic again, because it's also a bug I can't seem to replicate as easily.

Reply:
> This means a new reaction is created or an existing reaction is destroyed, there's no in between in the controller.

This seems like we could be committing to something even more complex, but I trust your experience on this topic 😅 💪

Reply:
Moved this change out of this PR so that it's not blocking the other bug fix which is happening in `app/assets/javascripts/initializers/initializeCommentsPage.js.erb `